common/io/src/buffer/buffer_read_ext.rs

```diff
@@ -122,11 +122,20 @@ pub trait BufferReadExt: BufferRead {
     }
 
     fn read_escaped_string_text(&mut self, buf: &mut Vec<u8>) -> Result<()> {
-        self.keep_read(buf, |f| f != b'\t' && f != b'\n' && f != b'\\')?;
-        if self.ignore_byte(b'\\')? {
-            // TODO parse complex escape sequence
-            buf.push(b'\\');
-            self.read_escaped_string_text(buf)?;
+        loop {
+            self.keep_read(buf, |f| f != b'\t' && f != b'\n' && f != b'\\')?;
+            if self.ignore_byte(b'\\')? {
+                let buffer = self.fill_buf()?;
+                let c = buffer[0];
+                match c {
+                    b'\'' | b'\"' | b'\\' => {}
```

Comment:
Some other cases in [clickhouse](https://github.com/ClickHouse/ClickHouse/blob/aa737b7a11167fa4af3bb8f8dc4e70909d1728e2/src/IO/ReadHelpers.cpp):


1. `\x2A` to read hex codes
2. `\N`  empty string, skipped
3. ...

Reply:
ok,  these are not as important (at least not fail the loading),
I will add them after the tests passed

Reply:
done